lib/cea/cea608_data_channel.js

```diff
@@ -0,0 +1,720 @@
+   * @private
+   */
+  controlFon_() {
+    this.curbuf.addChar(
+        shaka.cea.Cea608Memory.CharSet.BASIC_NORTH_AMERICAN,
+        ' '.charCodeAt(0));
+  }
+
+
+  /**
+   * Handles EDM - Erase Displayed Mem
+   * Mode check:
+   * EDM affects all captioning modes (but not Text mode);
+   * @param {!number} pts
+   * @private
+   */
+  controlEdm_(pts) {
+    const buf = this.dpm_;
+    if (this.type_ !== shaka.cea.Cea608DataChannel.CaptionType.TEXT) {
+      // Clearing displayed memory means we now know how long
+      // its contents were displayed, so force it out.
+      buf.forceEmit(this.prevEndTime_, pts);
+    }
+    buf.resetAllRows();
+  }
+
+  /**
+   * Handles RDC - Resume Direct Captions. Initiates Paint-On captioning mode.
+   * RDC does not affect current display, so nothing needs to be forced out yet.
+   * @param {!number} pts in seconds
+   * @private
+   */
+  controlRdc_(pts) {
+    this.type_ = shaka.cea.Cea608DataChannel.CaptionType.PAINTON;
+    // Point to displayed memory.
+    this.curbuf = this.dpm_;
+
+    // No scroll window now.
+    this.curbuf.setScrollSize(0);
+
+    // The next paint-on caption needs this time as the start time.
+    this.prevEndTime_ = pts;
+  }
+
+
+  /**
+   * Handles ENM - Erase Nondisplayed Mem
+   * @private
+   */
+  controlEnm_() { this.ndm_.resetAllRows(); }
+
+  /**
+   * Handles EOC - End Of Caption (flip mem)
+   * This forces Pop-On mode, and swaps the displayed and nondisplayed memories.
+   * @private
+   * @param {!number} pts
+   */
+  controlEoc_(pts) {
+    this.dpm_.forceEmit(this.prevEndTime_, pts);
+    // Swap memories
+    const buf = this.ndm_;
+    this.ndm_ = this.dpm_;  // Swap buffers
+    this.dpm_ = buf;
+
+    // Enter Pop-On mode.
+    this.controlRcl_();
+
+    // The caption ended, and so the previous end time should be updated.
+    this.prevEndTime_ = pts;
+  }
+
+  /**
+   * Handles RCL - Resume Caption Loading
+   * Initiates Pop-On style captioning. No need to force anything out upon
+   * entering Pop-On mode because it does not affect the current display.
+   * @private
+   */
+  controlRcl_() {
+    this.type_ = shaka.cea.Cea608DataChannel.CaptionType.POPON;
+    this.curbuf = this.ndm_;
+    // No scroll window now
+    this.curbuf.setScrollSize(0);
+  }
+
+
+  /**
+   * Handles BS - BackSpace.
+   * @private
+   */
+  controlBs_() {
+    this.curbuf.eraseChar();
+  }
+
+  /**
+   * Handles TR - Text Restart.
+   * Clears text buffer and resumes Text Mode.
+   * @private
+   */
+  controlTr_() {
+    const buf = this.text_;
+
+    // Window for text mode is the entire screen i.e. all of the rows.
+    buf.setScrollSize(shaka.cea.Cea608Memory.CC_ROWS);
+    buf.eraseBuffer();
+    this.controlRtd_();  // Put into text mode.
+  }
+
+  /**
+   * Handles RTD - Resume Text Display.
+   * Resumes text mode. Mo need to force anything out, because Text Mode doesn't
+   * affect the current display. Also, this decoder does not emit TEXTn anyway.
```

Comment:
What is "TEXTn"?

Reply:
I've changed the confusing "TEXTn" to "text mode", and added a comment at the top of the class which describes the field "this.text_", and explains CEA-608 text mode.